services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -230,4 +236,85 @@ async def inspect_container(
         return inspect_result
 
 
+@containers_router.put(
+    "/containers:restore-state",
```

Comment:
- why not using state and nodeports as  sub-resources of containers? This will lead to
- Why are you using PUT instead of the expected POST?
```
POST /containers/state:restore
POST /containers/state:save
POST /containers/node-ports:pull
POST /containers/node-ports:push
```

PLEASE read https://cloud.google.com/apis/design/resources#design_flow

Reply:
@pcrespov, @GitHK  I would propose here even going like:
POST ```/containers/inputs:pull``` or ```/containers/ports:pull```
POST ```/containers/outputs:push``` or ```/containers/ports:push```

Reply:
I prefer the `/containers/ports:pull` & `/containers/ports:push`. I'd go for the following.

```
POST /containers/state:restore
POST /containers/state:save
POST /containers/ports:pull
POST /containers/ports:push
```

Reply:
be aware that a docker container also has ports and states ... and those are attributes. A bit of extra info in the name (whatever you find it best) would not harm